x-pack/plugins/lists/public/exceptions/transforms.ts

```diff
@@ -0,0 +1,110 @@
+ * in the form of:
+ * flow(addIdToExceptionItemEntries, myNewTransform)(exceptionItem)
+ *
+ * @param exceptionItem The exceptionItem to transform the output of
+ * @returns The exceptionItem transformed from the output
+ */
+export const transformInput = (exceptionItem: ExceptionListItemSchema): ExceptionListItemSchema =>
+  flow(addIdToExceptionItemEntries)(exceptionItem);
+
+/**
+ * This adds an id to the incoming exception item entries as ReactJS prefers to have
+ * an id added to them for use as a stable id. Later if we decide to change the data
+ * model to have id's within the array then this code should be removed. If not, then
+ * this code should stay as an adapter for ReactJS.
+ *
+ * This does break the type system slightly as we are lying a bit to the type system as we return
+ * the same exceptionItem as we have previously but are augmenting the arrays with an id which TypeScript
+ * doesn't mind us doing here. However, downstream you will notice that you have an id when the type
+ * does not indicate it. In that case use (ExceptionItem & { id: string }) temporarily if you're using the id. If you're not,
+ * you can ignore the id and just use the normal TypeScript with ReactJS.
+ *
+ * @param exceptionItem The exceptionItem to add an id to the threat matches.
+ * @returns exceptionItem The exceptionItem but with id added to the exception item entries
+ */
+export const addIdToExceptionItemEntries = (
+  exceptionItem: ExceptionListItemSchema
+): ExceptionListItemSchema => {
+  const entries = exceptionItem.entries.map((entry) => {
+    if (entry.type === 'nested') {
+      return addIdToItem({
+        ...entry,
+        entries: entry.entries.map((nestedEntry) => addIdToItem(nestedEntry)),
+      });
+    } else {
+      return addIdToItem(entry);
+    }
+  });
+  return { ...exceptionItem, entries };
+};
+
+/**
+ * This removes an id from the exceptionItem entries as ReactJS prefers to have
+ * an id added to them for use as a stable id. Later if we decide to change the data
+ * model to have id's within the array then this code should be removed. If not, then
+ * this code should stay as an adapter for ReactJS.
+ *
+ * @param exceptionItem The exceptionItem to remove an id from the entries.
+ * @returns exceptionItem The exceptionItem but with id removed from the entries
+ */
+export const removeIdFromExceptionItemsEntries = (
+  exceptionItem: CreateExceptionListItemSchema | UpdateExceptionListItemSchema
+): CreateExceptionListItemSchema | UpdateExceptionListItemSchema => {
+  const { entries, ...itemInfo } = exceptionItem;
+  const entriesNoId = entries.map((entry) => {
+    if (entry.type === 'nested') {
+      return removeIdFromItem({
+        ...itemInfo,
+        entries: entry.entries.map((nestedEntry) => removeIdFromItem(nestedEntry)),
+      });
+    } else {
+      return removeIdFromItem<Entry>(entry);
+    }
+  });
+  return { ...itemInfo, entries: entriesNoId as EntriesArray };
```

Comment:
I would really love to get rid of this `as` here, but I lost the battle. Any ideas, much appreciated.